editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/preferences/PreferredLabelDialog.java

```diff
@@ -0,0 +1,219 @@
+	private ComboViewer parameterCombo;
+
+	private IInputValidator componentValidator;
+	private IInputValidator parameterValidator;
+
+	private CamelModel camelModel;
+
+	public PreferredLabelDialog(Shell parentShell) {
+		super(parentShell);
+	}
+
+	public String getComponent() {
+		return component;
+	}
+
+	public void setComponent(String component) {
+		this.component = component;
+	}
+
+	public String getParameter() {
+		return parameter;
+	}
+
+	public void setParameter(String parameter) {
+		this.parameter = parameter;
+	}
+
+	public void setComponentValidator(IInputValidator componentValidator) {
+		this.componentValidator = componentValidator;
+	}
+
+	public void setParameterValidator(IInputValidator parameterValidator) {
+		this.parameterValidator = parameterValidator;
+	}
+
+	protected CamelModel getCamelModel() {
+		if (camelModel == null) {
+			camelModel = new CamelService().getCamelModel(getLatestCamelVersion(), RUNTIME_PROVIDER_KARAF);
+		}
+		return camelModel;
+	}
+
+	protected List<Eip> getComponents() {
+		return getCamelModel().getEips().stream().filter(eip -> !"NONE".equals(getCategoryFromEip(eip)))
+				.collect(Collectors.toList());
+	}
+
+	public void validate() {
+		String msg = null;
+		if (componentValidator != null) {
+			msg = componentValidator.isValid(componentCombo.getCombo().getText());
+		}
+		if (msg == null && parameterValidator != null) {
+			msg = parameterValidator.isValid(parameterCombo.getCombo().getText());
+		}
+		setWarning(msg);
+	}
+
+	public void setWarning(String msg) {
+		Button okButton = getButton(IDialogConstants.OK_ID);
+		if (okButton != null) {
+			setMessage(msg, msg == null ? IMessageProvider.NONE : IMessageProvider.WARNING);
+			okButton.setEnabled(msg == null);
+		}
+	}
+
+	@Override
+	public void create() {
+		super.create();
+		setTitle(UIMessages.preferredLabelsTitle);
+
+		componentCombo.addSelectionChangedListener(event -> {
+			ISelection selection = event.getSelection();
+			if (selection instanceof StructuredSelection) {
+				Object selectedObject = ((StructuredSelection) selection).getFirstElement();
+				if (selectedObject instanceof Eip) {
+					parameterCombo.setInput(((Eip) selectedObject).getParameters());
+				}
+			}
+		});
+
+		if (component != null && !component.isEmpty()) {
+			Eip eip = getCamelModel().getEip(component);
+			componentCombo.getCombo().setEnabled(false);
+			componentCombo.setInput(new Eip[] { eip });
+			componentCombo.setSelection(new StructuredSelection(eip));
+			if (parameter != null && !parameter.isEmpty()) {
+				parameterCombo.setSelection(new StructuredSelection(eip.getParameter(parameter)));
+			}
+		} else {
+			componentCombo.setInput(getComponents());
+		}
+
+		validate();
+	}
+
+	@Override
+	protected Control createDialogArea(Composite parent) {
+		Composite area = (Composite) super.createDialogArea(parent);
+		Composite container = new Composite(area, SWT.NONE);
+		container.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
```

Comment:
just FYI, i usually prefer to use GridDataFactory.fillDefaults().grab(true, true).create() it avoids me to remind about all the parameters but using the n"normal" constructor is perfectly fine.